src/qt/intro.cpp

```diff
@@ -214,7 +214,10 @@ bool Intro::pickDataDirectory()
             }
             dataDir = intro.getDataDirectory();
             try {
-                TryCreateDirectories(GUIUtil::qstringToBoostPath(dataDir));
+                if (TryCreateDirectories(GUIUtil::qstringToBoostPath(dataDir))) {
+                    // If a new data directory has been created, make wallets subdirectory too
+                    TryCreateDirectories(GUIUtil::qstringToBoostPath(dataDir) / "wallets");
```

Comment:
The logic data_dir + "/wallets" exists in a couple of places. Doesn't this also ignores -walletdir?

Reply:
This is just for creating the initial wallets directory. If it's not used it will stay empty.